frontend/web/src/app/components/physical-part/physical-part-edit/physical-part-edit.component.ts

```diff
@@ -0,0 +1,176 @@
+    this.physicalPartService.getPhysicalPart(id).subscribe(
+      (physicalPart) => {
+        this.physicalPart = physicalPart;
+        this.populate();
+      },
+    );
+  }
+
+  /* populate data */
+  private populate() {
+    this.constructForms();
+  }
+
+  private constructPhysicalPrintForm() {
+
+  }
+  /* init forms */
+  private constructForms() {
+    const fields = {
+      id: [ { value: (this.physicalPart && this.physicalPart.id ? this.physicalPart.id : ''), disabled: true },
+        Validators.compose([Validators.required])],
+    };
+
+    this.requiredFieldsForm = this.formBuilder.group(fields);
+    this.loaded = true;
+  }
+   /* save product instance */
+   save() {
+    /* convert relevant fields */
+    const id = this.physicalPart.id;
+    const data = this.requiredFieldsForm.value;
+    data.physicalPrintID = this.physicalPart.physicalPrintID;
+    data.magicsPartPairingID = this.physicalPart.magicsPartPairingID;
+    data.orderedPartID = this.physicalPart.orderedPartID;
+
+    const physicalPart: PhysicalPart = new PhysicalPart(this.requiredFieldsForm.value);
+    this.creating ? delete physicalPart['id'] : physicalPart.id = id;
+    if (this.creating) { // a new product
+      this.physicalPartService.createPhysicalPart(physicalPart).subscribe(
+        (data) => {
+          if (this.nav) { this.back(); }
+          this.changed.emit(data);
+        }, (error) => {
+          console.log(error);
+          this.errorService.showAlert(error.verobose_message_header, error.verbose_message);
+        },
+      );
+    } else {
+      this.physicalPartService.updatePhysicalPart(physicalPart).subscribe(
+        (data) => {
+          if (this.nav) { this.back(); }
+          this.changed.emit(data);
+        }, (error) => {
+          console.log(error);
+          this.errorService.showAlert(error.verobose_message_header, error.verbose_message);
+        },
+      );
+    }
+  }
+
+  cancel() {
+    if (this.nav) { this.back(); }
+    this.changed.emit(this.physicalPart);
+  }
+
+  back() {
+    this._location.back();
+  }
+
+  updateMagicsPartPairingIDs(value) {
+    this.physicalPart.physicalPrintID = value;
+    this.physicalPrintService.getPhysicalPrint(value).subscribe(
+      (physicalPrint) => {
+        this.physicalPrint = physicalPrint;
+        this.digitalPrintService.getDigitalPrint(this.physicalPrint.digitalPrintID).subscribe((digitalPrint) => {
+          this.digitalPrint = digitalPrint;
+          this.magicsPartIDs = Object.keys(this.digitalPrint.magicsPartPairing);
+        });
+    });
+  }
+
+  updateOrderedPartIDs(value) {
+    this.physicalPart.magicsPartPairingID = Number(getValueFromObject(this.digitalPrint.magicsPartPairing, value));
+    this.orderService.getOrdersByDigitalPartID(getValueFromObject(this.digitalPrint.magicsPartPairing, value)).subscribe(
+      (orders) => {
+        this.orders = orders;
+        this.orderedParts = [];
+        for (let i = 0; i < this.orders.length; ++i) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Expected a 'for-of' loop instead of a 'for' loop with this simple iteration [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS4138)